web/src/hooks/useGetCurrentUrl.ts

```diff
@@ -0,0 +1,19 @@
+import { useParams } from 'react-router-dom';
+import { RouteParameters, ZoneKey } from 'types';
+import { baseUrl } from 'utils/constants';
+
+export function useGetCurrentUrl({ zoneId }: { zoneId?: ZoneKey }) {
+  const {
+    urlTimeAverage,
+    urlDatetime,
+    zoneId: zoneIdParameter,
+  } = useParams<RouteParameters>();
+  const zId = zoneId || zoneIdParameter;
+  const url =
+    baseUrl +
+    (zId
+      ? `/zone/${zId}/${urlTimeAverage}/${urlDatetime || new Date().toISOString()}`
```

Comment:
This isn't floored to 00:00Z, when the URL is not present. We could try it with something like this:

`  const selectedDatetime = useAtomValue(selectedDatetimeIndexAtom);
  const targetDate = urlDatetime ?? new Date(selectedDatetime.datetime).toISOString();`
  
  and would be nice to keep zoneId variable naming over zId

Reply:
Updated, LMK if I need any additional changes!